src/homework/homeworks/myhashmap/MyHashMap.java

```diff
@@ -0,0 +1,142 @@
+package homework.homeworks.myhashmap;
+
+import java.util.Arrays;
+
+public class MyHashMap<K, V> {
+    public static final double LOAD_FACTOR = 0.75;
+    public static final int MAP_INITIAL_SIZE = 16;
+    private Node<K, V>[] array;
+    private int numberOfNodes;
+    private int bucketsOccupied;
+
+    public MyHashMap() {
+        array = new Node[MAP_INITIAL_SIZE];
+    }
+
+    static class Node<K, V> {
+        private K key;
+        private V value;
+        private Node<K, V> nextNode;
+
+        public Node(K key, V value) {
+            this.key = key;
+            this.value = value;
+        }
+
+        @Override
+        public String toString() {
+            return "Node{" +
+                    "key=" + key +
+                    ", value=" + value +
+                    ", nextNode=" + nextNode +
+                    '}';
+        }
+
+    }
+
+    public void put(K key, V value) {
+        if (bucketsOccupied / array.length >= LOAD_FACTOR) {
+            Node<K, V>[] arrayNew = new Node[array.length * 2];
+            for (int i = 0; i < array.length; i++) {
+                arrayNew[array[i].key.hashCode() % arrayNew.length] = array[i];
+            }
+            array = arrayNew;
+        }
+        int hashCodeOfNode = key.hashCode();
+        int numerOfBucket = hashCodeOfNode % array.length;
```

Comment:
hashCodeOfNode % array.length is done multiple times can be moved to separate method